src/inputs/NumberField.test.tsx

```diff
@@ -104,6 +104,20 @@ describe("NumberFieldTest", () => {
     expect(r.days()).toHaveValue("1 day");
   });
 
+  it("allows override of numberFormatOptions", async () => {
+    const r = await render(
+      <TestNumberField
+        label="Cost"
+        value={1200}
+        numFractionDigits={2}
+        numberFormatOptions={{ style: "currency", currency: "USD" }}
```

Comment:
Thought: I assume this is probably the use case you need, i.e. handling non-cents dollars? Wdyt about adding a `type=dollars` as well?